controllers/comments.controllers.js

```diff
@@ -0,0 +1,29 @@
+const { removeComment } = require("../models/comment.model");
+const { checkExists } = require("../utils/utils");
+
+exports.deleteComment = (req, res, next) => {
+  const { comment_id } = req.params;
+
+  const checkId = checkExists("comments", "comment_id", comment_id);
+
+  Promise.resolve(checkId)
+    .then((result) => {
+      if (result === 0) {
+        return Promise.reject({
+          status: 404,
+          msg: "Comment cant be found",
+        });
+      } else {
+        return;
+      }
+    })
+    .then(() => {
+      return removeComment(comment_id);
+    })
+    .then((result) => {
+      res.status(204).send({});
+    })
+    .catch((err) => {
+      next(err);
+    });
+};
```

Comment:
some unusual syntax here

if you want to `checkExists` then continue after that promise has resolved then you should chain a `.then` off of `checkExists` itself

```diff
- Promise.resolve(checkId).then(/*...*/)
+ checkExists("comments", "comment_id", comment_id).then(/*...*/)
```

make sure you are returning your promises!